src/main/java/com/amplitude/api/AmplitudeClient.java

```diff
@@ -465,7 +465,17 @@ public AmplitudeClient enableNewDeviceIdPerInstall(boolean newDeviceIdPerInstall
      * @return the AmplitudeClient
      */
     public AmplitudeClient useAdvertisingIdForDeviceId() {
-        this.useAdvertisingIdForDeviceId = true;
+        deviceIdType = DeviceIdType.ADVERTISING_ID;
+        return this;
+    }
+
+    /**
+     * Use Android app set id as the user's device ID.
+     *
+     * @return the AmplitudeClient
+     */
+    public AmplitudeClient useAppSetIdForDeviceId() {
```

Comment:
Does iOS SDK have this API for idfv? 
If it doesn't, does it make sense to have it here?

Reply:
iOS sdk is a bit different, it doesn't have an extra interface for this. Instead, it is using the tracking options and directly do a fallback after idfa. Do we want to align the preference by just use app set id as fallback and remove this method? iOS sdk does use both useAdvertisingIdForDeviceId and tracking option to track idfa as well while android doesn't use tracking option for device id judgement.

Reply:
I think this is actually better. You can use your own judgement for it.

Reply:
Sounds good! If we didn't use tracking options to judge device id, then we can follow the same pattern. It might be better to not have the tracking option and device id logic couple together.